anchore_engine/clients/grype_wrapper.py

```diff
@@ -455,3 +456,61 @@ def query_vulnerabilities(
             return query.all()
         finally:
             read_lock.release()
+
+
+@dataclass
+class RecordSource:
+    count: int
+    feed: str
+    group: str
+    last_synced: str
+
+
+def query_record_source_counts():
+    """
+    Query the current feed group counts for all current vulnerabilities.
+    """
+    # Get and release read locks
+    read_lock = grype_db_lock.gen_rlock()
+    if read_lock.acquire(blocking=False, timeout=60):
+        try:
+            logger.debug("Querying grype_db for feed group counts")
+
+            # Get the counts for each record source
+            results = (
+                _get_grype_db_session()
+                .query(
+                    GrypeVulnerability.record_source,
+                    func.count(GrypeVulnerability.record_source).label("count"),
+                )
+                .group_by(GrypeVulnerability.record_source)
+                .all()
+            )
+
+            # Get the timestamp from the current metadata file
+            metadata = get_current_grype_db_metadata()
+            last_synced = metadata["built"]
+
+            # Transform the results along with the last_synced timestamp for each result
+            output = []
+            for result in results:
+                feed_group = result[0].split(":", 1)
+                if len(feed_group) != 2:
+                    logger.error(
+                        "Unable to process feed/group for record_source {}. Omitting from the response".format()
+                    )
```

Comment:
Missing argument to be plugged into log statement. Also switch to safe logging format
```suggestion
                    logger.error(
                        "Unable to process feed/group for record_source %s. Omitting from the response", feed_group
                    )

Reply:
Thanks, will fix the missing argument. In the next PR I've already switched the logging format, so I'd rather just leave that for there though.